solarforecastarbiter/metrics/preprocessing.py

```diff
@@ -307,15 +357,17 @@ def filter_resample(fx_obs, fx_data, obs_data, ref_data, quality_flag):
     ref_fx = fx_obs.reference_forecast
 
     # raise ValueError if intervals don't match
+    # move this to align?
```

Comment:
If anything, I would make `_check_ref_fx` public and run that in `process_forecast_observations` early on (catching and logging as appropriate)

Reply:
`preprocessing.check_reference_forecast_consistency`